pennylane_lightning_gpu/src/simulator/MPIWorker.hpp

```diff
@@ -81,6 +82,45 @@ inline std::vector<int2> createWirePairs(const int numLocalQubits,
     return wirePairs;
 }
 
+/**
+ * @brief Create wire pairs for bit index swap and transform all target wires to
+ * local ones.
+ *
+ * @param numLocalQubits Number of local qubits.
+ * @param numTotalQubits Number of total qubits.
+ * @param tgts Vector of target wires.
+ * @return wirePairs Wire pairs to be passed to SV bit index swap worker.
+ */
+inline std::vector<int2> createWirePairs(int numLocalQubits, int numTotalQubits,
+                                         std::vector<int> &tgts,
+                                         std::vector<int> &statusWires) {
+    std::vector<int2> wirePairs;
+    int localbit = numLocalQubits - 1, globalbit = numLocalQubits;
+    while (localbit >= 0 && globalbit < numTotalQubits) {
+        if (statusWires[localbit] == WireStatus::Default &&
+            statusWires[globalbit] != WireStatus::Default) {
+            int2 wirepair = make_int2(localbit, globalbit);
+            wirePairs.push_back(wirepair);
+            if (statusWires[globalbit] == WireStatus::Target) {
+                for (size_t k = 0; k < tgts.size(); k++) {
+                    if (tgts[k] == globalbit) {
+                        tgts[k] = localbit;
+                    }
+                }
+            }
+            std::swap(statusWires[localbit], statusWires[globalbit]);
+        } else {
+            if (statusWires[localbit] != WireStatus::Default) {
+                localbit--;
+            }
+            if (statusWires[globalbit] == WireStatus::Default) {
+                globalbit++;
+            }
+        }
+    }
+    return wirePairs;
```

Comment:
So could we have something like
```suggestion
    return createWirePairs(numLocalQubits, numTotalQubits, {}, tgts, statusWires);
```

Reply:
That's a good idea! Let me double check and get back to you.